server/genquery2/include/irods/private/genquery2_ast_types.hpp

```diff
@@ -0,0 +1,278 @@
+
+        explicit condition_less_than_or_equal_to(std::string string_literal)
+            : string_literal{std::move(string_literal)}
+        {
+        }
+
+        std::string string_literal;
+    }; // struct condition_less_than_or_equal_to
+
+    struct condition_greater_than
+    {
+        condition_greater_than() = default;
+
+        explicit condition_greater_than(std::string string_literal)
+            : string_literal{std::move(string_literal)}
+        {
+        }
+
+        std::string string_literal;
+    }; // struct condition_greater_than
+
+    struct condition_greater_than_or_equal_to
+    {
+        condition_greater_than_or_equal_to() = default;
+
+        explicit condition_greater_than_or_equal_to(std::string string_literal)
+            : string_literal{std::move(string_literal)}
+        {
+        }
+
+        std::string string_literal;
+    }; // struct condition_greater_than_or_equal_to
+
+    struct condition_is_null
+    {
+    }; // struct condition_is_null
+
+    struct condition_is_not_null
+    {
+    }; // struct condition_is_not_null
+
+    struct condition_operator_not;
+
+    using condition_expression = boost::variant<condition_like,
+                                                condition_in,
+                                                condition_between,
+                                                condition_equal,
+                                                condition_not_equal,
+                                                condition_less_than,
+                                                condition_less_than_or_equal_to,
+                                                condition_greater_than,
+                                                condition_greater_than_or_equal_to,
+                                                condition_is_null,
+                                                condition_is_not_null,
+                                                boost::recursive_wrapper<condition_operator_not>>;
+
+    struct condition_operator_not
+    {
+        condition_operator_not() = default;
+
+        condition_operator_not(condition_expression expression)
+            : expression{std::move(expression)}
+        {
+        }
+
+        condition_expression expression;
+    }; // struct condition_operator_not
+
+    struct condition
+    {
+        condition() = default;
+
+        condition(column column, condition_expression expression)
+            : column{std::move(column)}
+            , expression{std::move(expression)}
+        {
+        }
+
+        column column;
+        condition_expression expression;
+    }; // struct condition
+
+    struct logical_and;
+    struct logical_or;
+    struct logical_not;
+    struct logical_grouping;
+
+    using condition_type = boost::variant<logical_and, logical_or, logical_not, logical_grouping, condition>;
+
+    // clang-format off
+    using selection  = boost::variant<select_function, column>;
```

Comment:
Leaving as is for now since the use of `_type` here is mostly due to `condition` already being used.

We can refactor later if it proves to be a challenge maintaining. I'll add a comment to explaining the "why" for future readers.

Reply:
Sure. Experimental, ahoy!

Reply:
Added comment to code for future readers.